torch/distributed/_sharded_tensor/api.py

```diff
@@ -590,6 +590,18 @@ def size(self) -> torch.Size:
         """
         return self._metadata.size
 
+    @property
+    def requires_grad(self):
+        return self._metadata.requires_grad
```

Comment:
nit: should we add the rest as well `pin_memory`, `memory_layout` etc.? Also, should `size` be `property` too?

Reply:
good point, I can add `is_pinned()` although that's not a property. I looked into tensor attributes page https://pytorch.org/docs/stable/tensor_attributes.html and tried locally, the local tensor doesn't have `.memory_layout` properties exposed, maybe we should only expose a `is_contiguous()`.

For `size`, iirc for local tensors, user can call `size()`, but couldn't do `.size`, instead they do `.shape`, so we might expose `shape` if we want?